src/main/webapp/creatives/maps/Logger.js

```diff
@@ -0,0 +1,53 @@
+
+import {generateSessionId} from './utils.js';
+
+/** A list of server routes that keep track of each interaction when fetched. */
+const URL = {
+  INITIALIZE: "initialize",
+  FIND_NEAREST_LOCATION: "find-location",
+  GRANT_LOCATION_DATA: "grant-location",
+  MAP_INTERACT: "map-interact",
+  SKIP_TO_CONTENT: "skip-to-content",
+  RETURN_TO_AD: "return-to-ad"
+}
+
+class Logger {
+  /**
+   * Each instance gets a correlator to uniquely identify it.
+   */
+  constructor() {
+    this.correlator = generateSessionId();
+  }
+
+  /**
+   * Makes a GET request to the given URL with the correlator as a parameter.
+   * @param {string} url A URL to send the request to.
+   */
+  sendRequest_(url) {
+    fetch(`/${url}?correlator="${this.correlator}"`);
```

Comment:
Did you look into the reportTracking SIMID message (https://interactiveadvertisingbureau.github.io/SIMID/#simid-creative-reportTracking)

I think this is also fine -- just wondering if you had a reason not to use it